### PR TITLE
mrc-204 adding new groups and members to those groups

### PR DESCRIPTION
**Pull request by hillalex**
Last bit of the CLI - creating new user groups and adding users to them.

### CODE REVIEW

File: src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/AuthorizationRepository.kt [hunk 1]
```diff
@@ -23,6 +23,43 @@ class OrderlyAuthorizationRepository : AuthorizationRepository
...
+        JooqContext().use {
+            it.dsl.selectFrom(ORDERLYWEB_USER_GROUP)
+                    .where(ORDERLYWEB_USER_GROUP.ID.eq(userGroup))
+                    .singleOrNull() ?: throw UnknownObjectError(userGroup, "user-group")
+
+            it.dsl.selectFrom(ORDERLYWEB_USER)
+                    .where(ORDERLYWEB_USER.EMAIL.eq(email))
+                    .singleOrNull() ?: throw UnknownObjectError(email, User::class)
+
+            it.dsl.newRecord(ORDERLYWEB_USER_GROUP_USER)
```
Comment (EmmaLRussell) [inline src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/AuthorizationRepository.kt]: Should this check that user isn't already a member of a group before doing the insert? 'Ensure' implies that it's fine if it is already (won't error) but we wouldn't want duplicate rows.

File: src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/database_tests/AuthorizationRepositoryTests.kt [hunk 2]
```diff
@@ -126,4 +155,41 @@ class OrderlyWebAuthorizationRepositoryTests : CleanDatabaseTests()
...
+        val sut = OrderlyAuthorizationRepository()
+        sut.createUserGroup("somegroup")
+        JooqContext().use {
+            it.dsl.insertInto(ORDERLYWEB_USER)
+                    .set(ORDERLYWEB_USER.EMAIL, "user@email.com")
+                    .set(ORDERLYWEB_USER.USER_SOURCE, "GitHub")
+                    .set(ORDERLYWEB_USER.USERNAME, "user.name")
+                    .execute()
+        }
+        sut.ensureGroupHasMember("somegroup", "user@email.com")
```
Comment (EmmaLRussell) [inline src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/database_tests/AuthorizationRepositoryTests.kt]: Could also check the user-group-user row was actually added to the db.
Oh, although I see you are checking that in the CLI tests..

File: src/userCLI/src/test/kotlin/org/vaccineimpact/orderlyweb/userCLI/tests/AddUserGroupTests.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.junit.Test
+import org.vaccineimpact.orderlyweb.db.JooqContext
+import org.vaccineimpact.orderlyweb.db.Tables.ORDERLYWEB_USER_GROUP
+import org.vaccineimpact.orderlyweb.test_helpers.CleanDatabaseTests
+import org.vaccineimpact.orderlyweb.userCLI.addUserGroup
+
+class AddUserGroupTests : CleanDatabaseTests()
+{
+    @Test
+    fun `addUser adds user`()
```
Comment (EmmaLRussell) [inline src/userCLI/src/test/kotlin/org/vaccineimpact/orderlyweb/userCLI/tests/AddUserGroupTests.kt]: Both tests should be named `addUserGroup etc...`?

File: src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/database_tests/AuthorizationRepositoryTests.kt (hunk 2 above)
Comment (hillalex) [reply src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/database_tests/AuthorizationRepositoryTests.kt]: Oh yeah, that should go here!

**Pull request merged**